Team9ProjectTests/src/com/indragie/comput301as1/test/SortTest.java

```diff
@@ -0,0 +1,112 @@
+package com.indragie.comput301as1.test;
+
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+
+import android.content.Intent;
+import android.test.AndroidTestCase;
+import android.util.Log;
+
+import com.indragie.cmput301as1.CreationDateAscendingComparator;
+import com.indragie.cmput301as1.CreationDateDescendingComparator;
+import com.indragie.cmput301as1.ExpenseClaim;
+import com.indragie.cmput301as1.ExpenseClaimSortActivity;
+import com.indragie.cmput301as1.ListModel;
+import com.indragie.cmput301as1.StartDateAscendingComparator;
+import com.indragie.cmput301as1.StartDateDescendingComparator;
+
+public class SortTest extends AndroidTestCase {
+	
+private static final String EXPENSE_CLAIM_SORT = null;
+	
+	@SuppressWarnings("unchecked")
+	public void testStart() {
+		// startDate in format (YYYY, MM, DD)
+		Date date1 = new Date(2000, 01, 01);
+		Date date2 = new Date(1995, 01, 01);
+		
+		// ExpenseClaim(name, description, startDate, endDate, status, creationDate)
+		ExpenseClaim claim1 = new ExpenseClaim(null, null, date1, null, null);
+		ExpenseClaim claim2 = new ExpenseClaim(null, null, date2, null, null);
+		
+		ListModel<ExpenseClaim> listmodel = new ListModel<ExpenseClaim>("claims.dat", getContext());
+		
+		/*
+		String test3 = claim1.getStartDate().toString();
+		String test2 = claim2.getStartDate().toString();		
+		Log.d(test3, test2);
+		*/
+		// add the list of claims to listmodel
+		listmodel.add(claim1);
+		listmodel.add(claim2);
+		
+		// get the list of claims
+		List<ExpenseClaim> claims = listmodel.getItems();
+		
+		// start the sorts
+		Intent data = getStartDateAscendingIntent();
+		listmodel.setComparator((Comparator<ExpenseClaim>) data.getSerializableExtra(ExpenseClaimSortActivity.EXPENSE_CLAIM_SORT));
+
+		assertNotNull(data);
```

Comment:
This check should be to see whether the result of `data.getSerializableExtra(ExpenseClaimSortActivity.EXPENSE_CLAIM_SORT)` is not null, not `data` itself (since it is clear that `data` will never be null).